src/configprovider.ts

```diff
@@ -1,54 +1,102 @@
 import * as vscode from "vscode";
-
+import * as nls from 'vscode-nls';
+import * as path from 'path';
 import {
-    getMesonTargets
+  getMesonTargets
 } from "./meson/introspection"
 import {
-    extensionConfiguration,
-    getTargetName
+  Target
+} from "./meson/types"
+import {
+  extensionConfiguration,
+  getTargetName
 } from "./utils"
 
+nls.config({ messageFormat: nls.MessageFormat.bundle, bundleFormat: nls.BundleFormat.standalone })();
+const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+
+export enum MIModes {
+  lldb = 'lldb',
+  gdb = 'gdb',
+}
+
 export class DebugConfigurationProvider implements vscode.DebugConfigurationProvider {
-    private path: string;
+  private path: string;
 
-    constructor(path: string) {
-        this.path = path
-    }
+  constructor(path: string) {
+    this.path = path
+  }
 
-    async provideDebugConfigurations(folder: vscode.WorkspaceFolder | undefined, token?: vscode.CancellationToken): Promise<vscode.DebugConfiguration[]> {
-        let targets = await getMesonTargets(this.path);
+  async createBaseDebugConfiguration(target: Target): Promise<vscode.DebugConfiguration> {
+    const targetName = await getTargetName(target);
+    return {
+      type: 'cppdbg',
+      name: `Debug ${target.name}`,
+      request: 'launch',
+      cwd: path.dirname(this.path),
+      program: target.filename[0],
+      args: [],
+      preLaunchTask: `Meson: Build ${targetName}`
+    };
+  }
 
-        let configDebugOptions = extensionConfiguration("debugOptions");
+  async createGDBDebugConfiguration(target: Target): Promise<vscode.DebugConfiguration> {
+    let debugConfig = await this.createBaseDebugConfiguration(target);
+    debugConfig.MIMode = MIModes.gdb;
+    debugConfig.setupCommands = [{
```

Comment:
I'm mildly convinced that I removed that on purpose because some doc says it may be an issue.

And there is a default config set that should merge in some setupCommands
Can you check the options and whether the startupCommands or debug thingy in there is considered for this?

Reply:
I tried replicating the same configuration used by CMake's extension https://github.com/microsoft/vscode-cmake-tools/blob/896f499b99a85e60b36bd8b8bf5a5e779af14236/src/debugger.ts#L94.

I don't know how to check if there is any config already provided, to be honest. I tried getting it from the context without luck.